target_chains/aptos/contracts/sources/pyth.move

```diff
@@ -163,20 +171,113 @@ module pyth::pyth {
+    fun verify_data_source(vaa: &vaa::VAA) {
         assert!(
             state::is_valid_data_source(
                 data_source::new(
-                    u16::to_u64(vaa::get_emitter_chain(&vaa)),
-                    vaa::get_emitter_address(&vaa))),
+                    u16::to_u64(vaa::get_emitter_chain(vaa)),
+                    vaa::get_emitter_address(vaa))),
             error::invalid_data_source());
+    }
+
+    fun parse_accumulator_merkle_root(message: vector<u8>): vector<u8> {
+        let msg_payload_cursor = cursor::init(message);
+        let payload_type = deserialize::deserialize_u32(&mut msg_payload_cursor);
+        assert!(payload_type == ACCUMULATOR_UPDATE_WORMHOLE_VERIFICATION_MAGIC, error::invalid_wormhole_message());
+        let wh_message_payload_type = deserialize::deserialize_u8(&mut msg_payload_cursor);
+        assert!(wh_message_payload_type == 0, error::invalid_wormhole_message()); // Merkle variant
+        let _merkle_root_slot = deserialize::deserialize_u64(&mut msg_payload_cursor);
+        let _merkle_root_ring_size = deserialize::deserialize_u32(&mut msg_payload_cursor);
+        let merkle_root_hash = deserialize::deserialize_vector(&mut msg_payload_cursor, 20);
+        cursor::rest(msg_payload_cursor);
+        merkle_root_hash
+    }
+
+    fun parse_accumulator_update_message(message: vector<u8>): PriceInfo {
+        let message_cur = cursor::init(message);
+        let message_type = deserialize::deserialize_u8(&mut message_cur);
+
+        assert!(message_type == 0, error::invalid_accumulator_message()); // PriceFeedMessage variant
+        let price_identifier = price_identifier::from_byte_vec(deserialize::deserialize_vector(&mut message_cur, 32));
+        let price = deserialize::deserialize_i64(&mut message_cur);
+        let conf = deserialize::deserialize_u64(&mut message_cur);
+        let expo = deserialize::deserialize_i32(&mut message_cur);
+        let publish_time = deserialize::deserialize_u64(&mut message_cur);
+        let _prev_publish_time = deserialize::deserialize_i64(&mut message_cur);
+        let ema_price = deserialize::deserialize_i64(&mut message_cur);
+        let ema_conf = deserialize::deserialize_u64(&mut message_cur);
+        let price_info = price_info::new(
+            timestamp::now_seconds(), // not used anywhere kept for backward compatibility
+            timestamp::now_seconds(),
+            price_feed::new(
+                price_identifier,
+                pyth::price::new(price, conf, expo, publish_time),
+                pyth::price::new(ema_price, ema_conf, expo, publish_time),
+            )
+        );
+        cursor::rest(message_cur);
+        price_info
+    }
 
-        // Deserialize the batch price attestation
-        update_cache(batch_price_attestation::destroy(batch_price_attestation::deserialize(vaa::destroy(vaa))));
+    fun parse_accumulator_updates(cursor: &mut Cursor<u8>, merkle_root: &vector<u8>): vector<PriceInfo> {
```

Comment:
i think it's good to prefix some of the methods with parse_and_verify_... for more clarity.